sdk/core/azure-core/inc/response.hpp

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <http/http.hpp>
+
+namespace Azure { namespace Core {
+
+  template <class T> class Response {
+    T m_value;
+    std::unique_ptr<Http::RawResponse> m_rawResponse;
+
+  public:
+    // Require a raw response to create a Response T
+    Response(const T& initialValue, std::unique_ptr<Http::RawResponse> rawResponse)
+        : m_value(initialValue), m_rawResponse(std::move(rawResponse))
+    {
+    }
+
+    ~Response() { this->m_value.~T(); }
```

Comment:
This is wrong: `m_value` is a member so its dtor is called automatically so you're double destroying here. (An explicit dtor call would only be necessary if this was a union or similar.

Reply:
This memory corruption bug is the main reason I'm 'request changes' at the moment.